snowflakeid/src/main/java/snowflakeid/SnowflakeIdGenerator.java

```diff
@@ -0,0 +1,75 @@
+package snowflakeid;
+
+import java.time.Instant;
+
+public class SnowflakeIdGenerator {
+
+    private final long defaultEpoch = 1759536000000L;
+    private final long machineIdBits = 10L;
+    private final long maxMachineId = ~(-1L << machineIdBits);
+    private final long sequenceBits = 12L;
+
+    private long machineId;
+
+    public long getSequence() {
+        return sequence;
+    }
+
+    public void setSequence(long sequence) {
+        this.sequence = sequence;
+    }
+
+    private long sequence = -1L;
+
+    private long lastTimestamp = -1L;
+
+    public SnowflakeIdGenerator(long machineId) {
+        if (machineId > maxMachineId || machineId < 0) {
+            throw new IllegalArgumentException(String.format("Machine Id can't be greater than %d or less than 0", maxMachineId));
+        }
+        this.machineId = machineId;
+    }
+
+    public synchronized long nextId(Instant instant) {
```

Comment:
`synchronized` is very inefficient in this case.
Passing `Instant` makes no sense here, why would a Java library relies on the client code to generate current timestamp ?